### ISSUE TITLE
BatText.nsplit does not favour lack of spaces.

### ISSUE THREAD

**Issue by xenocons** [ocaml-batteries-team/batteries-included#561]
If there is no space in the string, nsplit does not work.

```
> nsplit (of_string "1,2,3,4") (of_string ",") |> List.map to_string;;
string list = ["1"]   
> nsplit (of_string "1, 2, 3, 4, ") (of_string ",") |> List.map to_string;;
string list = ["1"; " 2"; " 3"; " 4"]

**Comment by UnixJunkie** [comment]
Is this a bug ?

**Comment by gasche** [comment]
It looks like a bug indeed. If you feel motivated to investigate (for example, why hasn't the test suite caught the problem?) and propose a fix, that would be warmly welcome.

(I'm thinking of trying to get a Batteries release out, because people ask (rightly) for more frequent bugfix releases.)

**Comment by UnixJunkie** [comment]
I'll try to add a unit test to catch this.
Then we should fix it before we release.

**Comment by UnixJunkie** [comment]
the unit test is in there:
https://github.com/ocaml-batteries-team/batteries-included/pull/645

**Comment by UnixJunkie** [comment]
@xenocons why not send a pull request with your proposal?
